csharp/src/Ice/SlicStream.cs

```diff
@@ -318,44 +281,21 @@ internal override void ReceivedReset(long errorCode)
 
         private bool ReleaseFlowControlCredit(bool notifyPeer = false)
         {
-            if (IsControl)
-            {
-                return true;
-            }
-
             // If the flow control credit is not already released, decreases the bidirectional or unidirectional
             // semaphore (for outgoing streams) / count (for incoming streams).
             if (Interlocked.CompareExchange(ref _flowControlCreditReleased, 1, 0) == 0)
             {
+                _socket.ReleaseFlowControlCredit(this);
+
                 if (IsIncoming)
                 {
-                    if (IsBidirectional)
-                    {
-                        Interlocked.Decrement(ref _socket.BidirectionalStreamCount);
-                    }
-                    else
-                    {
-                        Interlocked.Decrement(ref _socket.UnidirectionalStreamCount);
-                    }
-
                     if (notifyPeer)
```

Comment:
can we replace the double if by and condition now?
```
if (IsIncoming && notifyPeer){ ...}
```

Reply:
Yes, fixed!